OpenStack-accounting/sql/nova_get_accounting_data.sql

```diff
@@ -0,0 +1,161 @@
+CREATE PROCEDURE `get_accounting_data`(IN starttime datetime, IN endtime datetime)
+BEGIN
+/*
+   Generates accounting data from the Nova database on various resources used
+   Some further comments below for the trickier bits
+ */
+SELECT
+    IFNULL(i.availability_zone, 'nova') AS AvailabilityZone,
+    p.name AS Project,
+    pp.name AS Department,
+    it.name AS Flavor,
+    COUNT(i.uuid) as VMs,
+    @VMSeconds:=SUM(IF(i.created_at <= starttime /* Captures VMs which were created outside of the period deleted out of the period */
+            AND (i.deleted_at >= endtime
+            OR ISNULL(i.deleted_at)),
+        TIMESTAMPDIFF(SECOND,
+            starttime,
+            endtime),
+        IF(i.created_at <= starttime /* Captures VMs which were created before the period and deleted during the period */
+                AND i.deleted_at < endtime,
+            TIMESTAMPDIFF(SECOND,
+                starttime,
+                i.deleted_at),
+            IF(i.created_at > starttime /* Captures VMs which were created during the period and deleted outside the period */
+                    AND (i.deleted_at >= endtime
+                    OR ISNULL(i.deleted_at)),
+                TIMESTAMPDIFF(SECOND,
+                    i.created_at,
+                    endtime),
+                TIMESTAMPDIFF(SECOND,
+                    i.created_at,
+                    i.deleted_at))))) AS VM_Seconds, /* Generates a count of seconds VMs were running */
+    'something' AS testingstuff,
```

Comment:
Is this for testing the procedure only? I think it should be removed from the query

Reply:
Yes it is, removed now